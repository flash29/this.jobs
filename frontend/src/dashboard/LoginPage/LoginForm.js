import React, { Component } from 'react';
import { useState} from 'react';
import {Form, FormGroup, FormControl, Button } from 'react-bootstrap';
import { useNavigate } from "react-router-dom";
import {H} from 'react-headings';
import './LoginPage.css'



function LoginForm(props) {

    const [postData, setPostData] = useState({ useremail : '', password : ''});

    const handleLoginClick = () =>{
        console.log(postData);
        fetch('/auth/login', {
            method : 'POST', 
            headers:{'Content-type':'application/json'},
            body:JSON.stringify(postData),
        }).then(response => response.json()).then(data => {
            console.log(data);
            sessionStorage.setItem(data.userEmail, data.password);
            setPostData({ useremail : '', password : ''});
            //window.location.reload(false)
            let path = "/home";
            navigate(path);
        }).catch(error => console.log('error', error))
    }

    let navigate = useNavigate();
    const handleRegisterClick = () => {
        let path = "/registration";
        navigate(path);
    }

    return (
        <div className = "divStyle">
            <H className = "title"> this.jobs</H>
            <Form className="LoginForm" id="loginForm">
                <FormGroup controlId="formEmail" className = "FormComp w-50">
                    <FormControl type="email" placeholder="Email Address" className = "inpBox" onChange={(e) => setPostData({...postData, useremail : e.target.value}) }/>
                </FormGroup>
                <FormGroup controlId="formPassword" className = "FormComp w-50" onChange = {(e) => setPostData({...postData, password : e.target.value})}>
                    <FormControl type="password" className = "FormComp" placeholder="Password" className = "inpBox"/>
                </FormGroup>
                <FormGroup controlId="formSubmit" className = "FormComp">
                    <Button  className = "buttonStyle" onClick={handleLoginClick}>
                        Login
                    </Button>
                    <Button  className = "buttonStyle"  onClick={handleRegisterClick}>
                        Register
                    </Button>
                </FormGroup>
            </Form>
        </div>
    )
}


export default LoginForm;
// bsstyle="primary" type="submit