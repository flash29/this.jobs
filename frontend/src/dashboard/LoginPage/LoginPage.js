import React, { Component } from 'react';
import LoginForm from './LoginForm';
import './LoginPage.css';

class LoginPage extends Component{
    render() {
        return (
            <div className = "LoginPage" data-testid = "LoginPage">
                <LoginForm data-testid = "loginForm"/>
            </div>
        );
    }
}

export default LoginPage;