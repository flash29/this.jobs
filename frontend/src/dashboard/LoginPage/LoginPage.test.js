import { render, screen, fireEvent } from '@testing-library/react';
import LoginPage from './LoginPage';
import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import LoginForm from './LoginForm';
import "@testing-library/jest-dom/extend-expect";
import { shallow, mount, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer';
import {createMemoryHistory} from 'history'
//const renderer = require('react-test-renderer'); 
//import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

test('Contains title', () => {
  render(
  <BrowserRouter>
    <LoginPage />
  </BrowserRouter>);
  expect(screen.getByText(/this.jobs/i)).toBeInTheDocument();
});

test("Email input field working", () => {
    render(
        <BrowserRouter>
          <LoginPage />
        </BrowserRouter>);
    expect(screen.queryByPlaceholderText(/Email Address/i)).toBeInTheDocument();
});

test("password input field working", () => {
    render(
        <BrowserRouter>
          <LoginPage />
        </BrowserRouter>);
    expect(screen.queryByPlaceholderText(/Password/i)).toBeInTheDocument();
});

test('Login Button test', () => {
    const {debug, getByText} = render(<BrowserRouter>
        <LoginPage />
      </BrowserRouter>);
    expect(getByText('Login').tagName).toBe('BUTTON')
    fireEvent.click(getByText('Login'))
    //debug()
});


test('Registration Button test', () => {
    const {debug, getByText} = render(<BrowserRouter>
        <LoginPage />
      </BrowserRouter>);
    expect(getByText('Register').tagName).toBe('BUTTON')
    fireEvent.click(getByText('Register'))
    //debug()
});

test('render correctly the LoginPage component', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <LoginPage />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

test('render correctly the LoginForm component', () => {  
    const TextInputComponent = renderer.create(<BrowserRouter>
        <LoginForm />
      </BrowserRouter>).toJSON();
    expect(TextInputComponent).toMatchSnapshot();
});

test('Sending the post data to /auth/login', () => {
  const history = createMemoryHistory()
  const route = '/auth/register'
  history.push(route)
  render(
    <BrowserRouter history={history}>
      <LoginPage />
    </BrowserRouter>)
});

test('Redirecting to register page ', () => {
  const history = createMemoryHistory()
  const route = '/registration'
  history.push(route)
  render(
    <BrowserRouter history={history}>
      <LoginPage />
    </BrowserRouter>)
});

test('Redirecting to home page ', () => {
  const history = createMemoryHistory()
  const route = '/home'
  history.push(route)
  render(
    <BrowserRouter history={history}>
      <LoginPage />
    </BrowserRouter>)
});
