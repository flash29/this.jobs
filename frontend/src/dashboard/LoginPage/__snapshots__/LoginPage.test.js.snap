// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly the LoginForm component 1`] = `
<div
  className="divStyle"
>
  <h1
    className="title"
  >
     this.jobs
  </h1>
  <form
    className="LoginForm"
    id="loginForm"
  >
    <div
      className="FormComp w-50"
    >
      <input
        className="inpBox form-control"
        id="formEmail"
        onChange={[Function]}
        placeholder="Email Address"
        type="email"
      />
    </div>
    <div
      className="FormComp w-50"
      onChange={[Function]}
    >
      <input
        className="FormComp inpBox form-control"
        id="formPassword"
        placeholder="Password"
        type="password"
      />
    </div>
    <div
      className="FormComp"
      data-testid="formSubmit"
    >
      <button
        className="buttonStyle btn btn-primary"
        data-testid="login"
        disabled={false}
        id="loginButton"
        onClick={[Function]}
        type="button"
      >
        Login
      </button>
      <button
        className="buttonStyle btn btn-primary"
        data-testid="reg"
        disabled={false}
        id="registrationButton"
        onClick={[Function]}
        type="button"
      >
        Register
      </button>
    </div>
  </form>
</div>
`;

exports[`render correctly the LoginPage component 1`] = `
<div
  className="LoginPage"
  data-testid="LoginPage"
>
  <div
    className="divStyle"
    data-testid="loginForm"
  >
    <h1
      className="title"
    >
       this.jobs
    </h1>
    <form
      className="LoginForm"
      id="loginForm"
    >
      <div
        className="FormComp w-50"
      >
        <input
          className="inpBox form-control"
          id="formEmail"
          onChange={[Function]}
          placeholder="Email Address"
          type="email"
        />
      </div>
      <div
        className="FormComp w-50"
        onChange={[Function]}
      >
        <input
          className="FormComp inpBox form-control"
          id="formPassword"
          placeholder="Password"
          type="password"
        />
      </div>
      <div
        className="FormComp"
        data-testid="formSubmit"
      >
        <button
          className="buttonStyle btn btn-primary"
          data-testid="login"
          disabled={false}
          id="loginButton"
          onClick={[Function]}
          type="button"
        >
          Login
        </button>
        <button
          className="buttonStyle btn btn-primary"
          data-testid="reg"
          disabled={false}
          id="registrationButton"
          onClick={[Function]}
          type="button"
        >
          Register
        </button>
      </div>
    </form>
  </div>
</div>
`;
